packages/db/src/lib/granule.ts

```diff
@@ -354,3 +357,39 @@ export const getGranulesByGranuleId = async (
     .where({ granule_id: granuleId });
   return records;
 };
+
+/**
+ * Change a granules' PG record and its files' PG record based on collection move
+ *
+ * @param {Knex | Knex.Transaction} knexOrTransaction - DB client or transaction
+ * @param {Array<ApiGranule>} [granules] - updated ApiGranule records
+ * @returns {Promise<void>}
+ */
+export const updateGranuleAndFiles = async (
+  knexOrTransaction: Knex | Knex.Transaction,
+  granules: Array<ApiGranule>
+):Promise<void> => {
+  const {
+    granules: granulesTable,
+    files: filesTable,
+  } = TableNames;
+  await Promise.all(granules.map(async (granule) => {
+    const pgGranule = await translateApiGranuleToPostgresGranule({
+      dynamoRecord: granule,
+      knexOrTransaction,
+    });
+    await knexOrTransaction(granulesTable).where('granule_id', '=', pgGranule.granule_id).update(
```

Comment:
I think we need to enforce a transaction here, not just have it be possible it's a transaction object.       We don't want partial granule/file record updates.

Reply:
@Nnaga1  I'd read up on https://knexjs.org/guide/transactions.html / take a look at what we're doing in the API logic for https://github.com/nasa/cumulus/blob/CUMULUS-3757-move-granule/packages/api/lib/writeRecords/write-granules.js#L488

Reply:
To be clear, I'm suggesting that we commit all granule/file updates together (and roll back on failure) in a transaction instead of make updates serially in a way that doesn't rollback if any of them fail.